src/control/build/interop_test.go

```diff
@@ -160,6 +160,37 @@ func TestBuild_CheckCompatibility(t *testing.T) {
 				},
 			},
 		},
+		"custom matcher: narrowly-scoped rule (non-match)": {
+			a: testComponent(t, "server", "2.0.0"),
+			b: testComponent(t, "agent", "2.0.0"),
+			customRule: &build.InteropRule{
+				Self:        build.ComponentServer,
+				Other:       build.ComponentAgent,
+				Description: "server/agent 2.0.0/2.0.1 interop rule",
+				Match: func(self, other *build.VersionedComponent) bool {
```

Comment:
Could these Match rules be set up to return true or false directly, to simplify things? The comparisons don't really seem to be meaningful here.

Reply:
... yes. I yanked that out of the updated rule from the 2.2 branch, but you're right, there's no reason to make the test more complicated than necessary.